src/OMSimulatorLib/DirectedGraph.cpp

```diff
@@ -277,3 +282,43 @@ void oms::DirectedGraph::calculateSortedConnections()
 
   sortedConnectionsAreValid = true;
 }
+
+/*
+ * Function which implements the allowed connections, according to SSP-1.0 connection table
+ */
+bool oms::DirectedGraph::isValidConnection(const ComRef& crefA, const ComRef& crefB, Connector& conA, Connector& conB)
```

Comment:
Why did you define `crefA` and `crefB` as an extra argument? This information is already present in `conA` and `conB`.

Reply:
This is needed, because the crefs can only detect  if it is top level System or Element, the connectors.getName(), does not give the fullcrefs, it just has only the connectors name